### Question: Is the new bus route for line X8 improving commute time (travel time at peak hours) ?

### Null Hypothesis H<sub>0</sub> and  Alternative Hypothesis H<sub>a</sub>--
H<sub>0</sub>: The average commute time with the new bus route is the same or longer, compared to that with the old bus route. i.e., T_new_avg >= T_old_avg

H<sub>a</sub>: The average commute time with the new bus route is shorter, compared to that with the old bus route. i.e., T_new_avg < T_old_avg

### Information on old travel times:
T_old ~ N(36, 36)

### Statistical test chosen:
Z-test (the number of standard deviations away from the mean)
$ Z = \frac{\mu_{pop} - \mu_{sample}}{\sigma / \sqrt{N}} $

In [1]:
import numpy as np

In [2]:
alpha = 0.05 #significance threshold chosen

In [3]:
!curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt
#new travel time measured.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
107  1179  107  1179    0     0   6813      0 --:--:-- --:--:-- --:--:-- 43666


In [4]:
T_new = np.loadtxt("times.txt") #module learned from SciPy.org

In [5]:
type(T_new)

numpy.ndarray

In [6]:
N = np.size(T_new)
N

100

In [7]:
T_new_avg = np.mean(T_new)
T_new_avg

34.466161688299998

In [8]:
T_old_avg = 36
sigma = 6

In [9]:
z_score = (T_old_avg - T_new_avg) / (sigma / (np.sqrt(N)))
z_score

2.5563971861666701

### Judgment: 2.56 > 2, meaning we are father than 2 sigmas away from the old mean. Two sigmas is 5%, equal to our chosen significance threshold. Which also means the p-value here is smaller than 0.05.

### Conclusion: The Null hypothesis is rejected. The average commute time with the new bus route is shorter. i.e., new route has brought improvement.